src/NLog/Internal/PropertyHelper.cs

```diff
@@ -221,15 +225,15 @@ private static bool TryNLogSpecificConversion(Type propertyType, string value, o
             return false;
         }
 
-        private static bool TryGetEnumValue(Type resultType, string value, out object result)
+        private static bool TryGetEnumValue(Type resultType, string value, out object result, bool flagsEnumAllowed)
         {
             if (!resultType.IsEnum)
             {
                 result = null;
                 return false;
             }
 
-            if (resultType.IsDefined(typeof(FlagsAttribute), false))
+            if (flagsEnumAllowed && resultType.IsDefined(typeof(FlagsAttribute), false))
```

Comment:
What if it's a flags-enum, but `flagsEnumAllowed` is false? The else-part would then execute with the flags-enum.